autoprecompiles/src/constraint_optimizer.rs

```diff
@@ -292,10 +294,9 @@ fn variables_in_stateful_bus_interactions<'a, P: FieldElement, V: Ord + Clone +
 fn remove_trivial_constraints<P: FieldElement, V: PartialEq + Clone + Hash + Ord>(
     mut constraint_system: JournalingConstraintSystem<P, V>,
 ) -> JournalingConstraintSystem<P, V> {
-    let zero = GroupedExpression::zero();
-    constraint_system.retain_algebraic_constraints(|constraint| constraint != &zero);
+    constraint_system.retain_algebraic_constraints(|constraint| !constraint.is_zero());
```

Comment:
I think here it would make sense to have `is_trivial` or `is_redundant` or something like that

Reply:
done